pkgs/applications/virtualization/qemu/default.nix

```diff
@@ -57,9 +58,22 @@ stdenv.mkDerivation (finalAttrs: {
     + lib.optionalString toolsOnly "-utils";
   version = "8.2.4";
 
-  src = fetchurl {
+  # Upstream doesn't vendor any dependencies in their release tarball,
+  # which for example breaks during cross-compilation.
+  # So instead we just download all subprojects.
+  src = fetchzip {
     url = "https://download.qemu.org/qemu-${finalAttrs.version}.tar.xz";
-    hash = "sha256-7PVTf+q5JkG5nXSC9VHyGV06W9NKzvnVK/v/NTpgc5c=";
+    hash = "sha256-PY/I4qCQj6Q0n8iFzQSHMGimMk0T6o8QAzUGQjF70YI=";
+    nativeBuildInputs = [ cacert git meson ];
+    postFetch = ''
+      (
+        cd "$out"
+        for prj in subprojects/*.wrap; do
```

Comment:
According to the error message in the upstream `configure` script ( https://github.com/qemu/qemu/blob/v8.2.4/configure#L782 ) it should be enough to invoke `meson subprojects download` once in the top-level folder. This has worked well for me and then we would not need this for-loop. Does that work for your case as well?

Reply:
In FODs it's important to ensure we depend on the behaviour of fetcher programs as little as possible, because an update to e.g. Meson that changes how it lays things out would break hashes.  So it's best to do as much ourselves as possible.

Reply:
meson could be moved out, but we still need to remove `.git` as it is not binary reproducible.